lib/thredded/database_seeder.rb

```diff
@@ -0,0 +1,269 @@
+        .update!(read_at: topic.updated_at, page: 1)
+    end
+
+    class BaseSeedData
+      attr_reader :seeder
+
+      def initialize(seed_database)
+        @seeder = seed_database
+      end
+
+      delegate :log, to: :seeder
+
+      def model_class
+        self.class::MODEL_CLASS
+      end
+
+      delegate :exists?, to: :model_class
+
+      def create(*_args)
+        fail 'override in your class!'
+      end
+
+      def find
+        model_class.all.to_a
+      end
+
+      def find_or_create(*args)
+        return @stored if @stored
+        @stored = if exists?
+                    find
+                  else
+                    create(*args)
+                  end
+      end
+    end
+
+    class FirstSeedData < BaseSeedData
+      def exists?
+        @stored ||= model_class.first
+      end
+
+      def find
+        @stored ||= model_class.first
+      end
+    end
+
+    class FirstUser < FirstSeedData
+      MODEL_CLASS = User
+
+      def create
+        log 'Creating first user...'
+        FactoryGirl.create(:user, :approved, :admin, name: 'Joe', email: 'joe@example.com')
+      end
+    end
+
+    class Users < BaseSeedData
+      MODEL_CLASS = User
+
+      def create(count: 1)
+        log "Creating #{count} users..."
+        approved_users_count = (count * 0.97).round
+        [seeder.first_user] +
+          FactoryGirl.create_list(:user, approved_users_count, :approved) +
+          FactoryGirl.create_list(:user, count - approved_users_count)
+      end
+    end
+
+    class FirstMessageboard < FirstSeedData
+      MODEL_CLASS = Messageboard
+
+      def create
+        log 'Creating a messageboard...'
+        @first_messageboard = FactoryGirl.create(
+          :messageboard,
+          name: 'Main Board',
+          slug: 'main-board',
+          description: 'A board is not a board without some posts'
+        )
+      end
+    end
+
+    class Topics < BaseSeedData
+      MODEL_CLASS = Topic
+
+      def create(count: 1, messageboard: seeder.first_messageboard)
+        log "Creating #{count} topics in #{messageboard.name}..."
+        FactoryGirl.create_list(
+          :topic, count,
+          messageboard: messageboard,
+          user: seeder.users.sample,
+          last_user: seeder.users.sample
+        )
+      end
+    end
+
+    class PrivateTopics < BaseSeedData
+      MODEL_CLASS = PrivateTopic
+
+      def create(count: 1)
+        FactoryGirl.create_list(
+          :private_topic, count,
+          user: seeder.users[1..-1].sample,
+          last_user: seeder.users.sample,
+          users: [seeder.first_user]
+        )
+      end
+    end
+
+    class Posts < BaseSeedData
+      MODEL_CLASS = Post
+
+      def create(count: (1..1))
+        log "Creating #{count} additional posts in each topic..."
+        seeder.topics.flat_map do |topic|
+          posts = []
+          written = (1 + rand(5)).days.ago
+          (count.min + rand(count.max + 1)).times do
```

Comment:
ok, rubocop then demanded I use `Array.new` with a block rather than `times.map` (seems less clear to me but ...)